src/routes/controllers/users.controller.js

```diff
@@ -216,3 +216,53 @@ export async function getUsersComments(req, res) {
     return errorResponse(res, 500, error.message);
   }
 }
+
+/**
+ *
+ * @param {Object} req express request
+ * @param {Object} res express response
+ * @returns {Array} user likes
+ */
+export async function getAllUserLikes(req, res) {
+  const { user } = req;
+  const { type } = req.query;
+
+  try {
+    const currentUser = await getUserbyId(user.id);
+
+    if (type === 'articles' || type === 'article') {
+      const allArticlesLikedByUser = await currentUser.getLikes();
+      return successResponse(
+        res,
+        200,
+        'Successfully retrieved all articles liked by user',
+        { articles: allArticlesLikedByUser },
+      );
+    } else if (type === 'comments' || type === 'comment') {
+      const allCommentsLikedByUser = await CommentLike.findAll({
+        where: { userId: user.id },
+      });
+      return successResponse(
+        res,
+        200,
+        'Successfully retrieved all comments liked by user',
+        { comments: allCommentsLikedByUser },
+      );
+    }
+    const allArticlesLikedByUser = await currentUser.getLikes();
+    const allCommentsLikedByUser = await CommentLike.findAll({
```

Comment:
And here also, we are duplicating line 242? Do you think there's a way we can make it DRY, perhaps using let instead of const